### PR TITLE
feat(api-client): catch OpenAPI parsing errors

### PR DESCRIPTION
**Pull request by hanspagel**
Currently, `createClient` and `createApiClientModal` throw an error for invalid OpenAPI documents.

With this PR those errors are catched and a helpful message is printed to the console.

Screenshot 2024-11-20 at 11 22 55

### CODE REVIEW

File: packages/openapi-parser/src/utils/makeFilesystem.test.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import { describe, expect, it } from 'vitest'
 
-import { getEntrypoint } from './getEntrypoint'
```
Comment (hanspagel) [inline packages/openapi-parser/src/utils/makeFilesystem.test.ts:3]: wasn’t used

**Review by tmastrom — APPROVED** [review]
Nice work!

**Review by geoffgscott — CHANGES_REQUESTED** [review]
This would be much more streamlined if we just caught the errors as needed in `importSpecFile`

For this particular case i think we should only catch the parsing errors and then log a suitable message to the console using `console.warn`

**Comment by geoffgscott** [comment]
Closing in favour of https://github.com/scalar/scalar/pull/3969/files

**Comment by hanspagel** [comment]
Okay, we’ve got something better!

**Pull request closed without merging**